crates/matrix-sdk/src/oidc.rs

```diff
@@ -452,7 +427,80 @@ impl Oidc {
             &mut rand::thread_rng(),
         )?;
 
-        let code = (use_authorization_url)(url, validation_data.state.clone()).await?;
+        let state = validation_data.state.clone();
+
+        self.client.inner.oidc_validation_data.write().await.insert(state.clone(), validation_data);
+
+        Ok(OidcAuthenticationUrl { url, state })
+    }
+
+    /// Authorize the given scope by using the Authorization Code flow.
+    ///
+    /// This should be called after the registered client has been restored with
+    /// [`Oidc::restore_registered_client()`] and the URL returned by
+    /// `url_for_authorize_scope_with_authorization_code` has been presented.
```

Comment:
We can skip the part about `restore_registered_client`, and putting the whole name with the type surrounded by brackets will link to the other method in the docs.

Also we want to call it after the user was redirected to the redirect URI.